File: php/L07/automobile.php
<?php
class Automobile
{
	// this.$marca = $marca è implicito
	public function __construct(
		private $marca,
		private $modello,
		private $anno=2026
	) {}


	public function __get($nome) {
		return $this->$nome;
	}

	public function __set($nome, $valore) {
		$this->$nome = $valore;
	}
}

File: Java/2025-2026/Prj09_Autosalone/src/model/Automobile.java
package model;

public class Automobile extends Veicolo {

	private String marca;
	private String modello;
	
	public Automobile(String nome, String marca, String modello) {
		super(nome);
		this.marca = marca;
		this.modello = modello;
	}

	public String getMarca() {
		return marca;
	}

	public String getModello() {
		return modello;
	}

	@Override
	public String toString() {
		return "Automobile [marca=" + marca + ", modello=" + modello + "]";
	}
	
	

}
